### ISSUE TITLE
RadioGroup component

### ISSUE THREAD

**Issue by vbence86** [metal/metal-clay-components#101]
### Disclaimer
ClayRadio component renders a markup that violates accessibility standards when using multiple elements with the same `name` attribute. Rendering multiple ClayRadio elements with the same name must be wrapped by either a `<fieldset>` element or a container having `role="radiogroup"` applied on it. 

### Expected markup 
```html
<div role="radiogroup">
  <input type="radio" name="myInput" role="radio" value="1" />
  <input type="radio" name="myInput" role="radio" value="2" />
</div>
```

### Proposal
To create a ClayRadioGroup component that implements a functionality to incorporate various ClayRadio components that are intended to have the same `name` attribute and renders them inside of a wrapper that contains the required `role="radiogroup"`. 

```javascript
  new ClayRadioGroup(
  {
    items: [
      new ClayRadio( // .... ),
      new ClayRadio( // .... )
    ]
  },
  '#my-radio-group'
);
```

**Comment by carloslancha** [comment]
I think for the ClayRadioGroup case we should keep it simpler. Since we have the option `inline` to not wrap the `input` it should be the user the one who composes correctly the group, wrapping it from outside with a `<div role="radiogroup">`

Is the case, for example, of `ClayDropdownBase`. We're using there `ClayRadio` for `type: radio` items, so we need to change the API to accept something like:

```
new ClayDropdownBase(
{
   items: [
      items: [
         {
            active: true,
            checked: true,
            inputValue: 'myradio1Value',
            label: 'My Radio 1',
         },
         {
            inputValue: 'myradio2Value',
            label: 'My Radio 2',
         }
      ],
      inputName: 'myradiogroupname',
      type: 'radiogroup',
   ]
}
```

**Comment by vbence86** [comment]
I think it's a also good idea. We just need to understand the effort and implications of the various solutions and once the preferred one is agreed upon this issue can be picked up.

**Comment by carloslancha** [comment]
I'll take care of this tomorrow ;)

**Comment by carloslancha** [comment]
https://github.com/metal/metal-clay-components/pull/102

**Comment by pat270** [comment]
@vbence86 can you link the accessibility standard you're referring to?

**Comment by vbence86** [comment]
@pat270 Absolutely! Related standard definition: **WCAG 2.0 (A) Section 508**. TLDR;

To cut it short, all `<input type="radio" ...` with the same name must be wrapped by either `<fieldset>` element or an arbitrary element that incorporates `role="radiogroup"` attribute while the corresponding `<input type="radio" ...` elements all contain the `role="radio"` attribute. 

The following example shows a valid implementation that doesn't violate the referred rule.
```html
<div role="radiogroup">
  <input type="radio" name="myRadioInput" role="radio" />
  <input type="radio" name="myRadioInput" role="radio" />
</div>
```  

Fore more information: https://dequeuniversity.com/rules/axe/1.1/radiogroup

**Comment by pat270** [comment]
Nice! Thanks @vbence86

**Comment by carloslancha** [comment]
@vbence86 can we close this issue? I guess it was solved in https://github.com/metal/metal-clay-components/pull/102 😺

**Comment by vbence86** [comment]
@carloslancha Yes, the reported issue has been fixed and this therefore can be closed.
